File: core-examples/src/main/java/io/vertx/example/core/execblocking/ExecBlockingExample.java
package io.vertx.example.core.execblocking;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Launcher;

/*
 * @author <a href="http://tfox.org">Tim Fox</a>
 */
public class ExecBlockingExample extends AbstractVerticle {

  public static void main(String[] args) {
    Launcher.executeCommand("run", ExecBlockingExample.class.getName());
  }

  @Override
  public void start() throws Exception {

    vertx.createHttpServer().requestHandler(request -> {

      // Let's say we have to call a blocking API (e.g. JDBC) to execute a query for each
      // request. We can't do this directly or it will block the event loop
      // But you can do this using executeBlocking:

      vertx.<String>executeBlocking(promise -> {

        // Do the blocking operation in here

        // Imagine this was a call to a blocking API to get the result
        try {
          Thread.sleep(500);
        } catch (Exception ignore) {
        }
        String result = "armadillos!";

        promise.complete(result);

      }, res -> {

        if (res.succeeded()) {

          request.response().putHeader("content-type", "text/plain").end(res.result());

        } else {
          res.cause().printStackTrace();
        }
      });

    }).listen(8080);

  }
}

File: core-examples/src/main/ruby/io/vertx/example/core/execblocking/exec_blocking_example.rb

$vertx.create_http_server().request_handler() { |request|

  # Let's say we have to call a blocking API (e.g. JDBC) to execute a query for each
  # request. We can't do this directly or it will block the event loop
  # But you can do this using executeBlocking:

  $vertx.execute_blocking(lambda { |promise|

    # Do the blocking operation in here

    # Imagine this was a call to a blocking API to get the result
    begin
      Java::JavaLang::Thread.sleep(500)
    rescue
    end

    result = "armadillos!"

    promise.complete(result)

  }) { |res_err,res|

    if (res_err == nil)

      request.response().put_header("content-type", "text/plain").end(res)

    else
      res_err.print_stack_trace()
    end
  }

}.listen(8080)

